addons/mail/static/src/core/common/message_reaction_list.js

```diff
@@ -0,0 +1,58 @@
+/* @odoo-module */
+
+import { Component } from "@odoo/owl";
+import { _t } from "@web/core/l10n/translation";
+
+export class MessageReactionList extends Component {
+    static template = "mail.MessageReaction.List";
+    static props = {
+        id: { type: Number, required: true },
+        close: { type: Function, required: true },
+        reaction: { type: Object, required: true },
+        openReactionMenu: { type: Function, required: true },
+    };
+
+    getReactionSummary() {
+        const reaction = this.props.reaction;
+        const [firstUserName, secondUserName, thirdUserName] = reaction.personas.map(
+            ({ name, displayName }) => name || displayName
+        );
```

Comment:
There's a `nameOrDisplayName` on persona model to do that.

Note that `displayName` exists due to `name` and `display_name` in python ORM being different things for a persona. It's not really desired to have both in Discuss scope, we'd like to keep only `name`. This is not doable yet, so right now we use `name` mostly and sometimes `nameOrDisplayName` when `display_name` is necessary.
Eventually, these `nameOrDisplayName` will be replaced to `name` when we finally remove the reliance on `display_name` in Discuss code.